### PR TITLE
Support year's quarters in format Q[1-4]

### PR DESCRIPTION
**Pull request by STayinloves**
GitHub issue #363

### CODE REVIEW

File: Specs/DateTime/English/DatePeriodExtractor.json [hunk 1]
```diff
@@ -2220,6 +2220,28 @@
...
+  "Results": [
+    {
+      "Text": "Q1",
+      "Type": "daterange"
+    }
+  ]
+}
+,
+{
+  "Input": "I'll be out this Q3",
```
Comment (tellarin) [inline Specs/DateTime/English/DatePeriodExtractor.json]: Can you add a case for "how did Q2 last year look like?"

File: Specs/DateTime/English/DatePeriodParser.json [hunk 2]
```diff
@@ -1748,6 +1748,81 @@
   ]
 }
 ,
+{
+  "Input": "I'll be out the Q3 this year",
```
Comment (tellarin) [inline Specs/DateTime/English/DatePeriodParser.json]: "I'll be out during Q3 this year" would be a more correct sentence.

File: Specs/DateTime/English/DatePeriodParser.json [hunk 3]
```diff
@@ -1748,6 +1748,81 @@
...
+          "startDate": "2016-07-01",
+          "endDate": "2016-10-01"
+        }
+      }
+    }
+  ]
+}
+,
+{
+  "Input": "I'll be out 2016 Q3",
```
Comment (tellarin) [inline Specs/DateTime/English/DatePeriodParser.json]: "Q1 2016" is a more common case, can you also add cases for it?

File: .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs [hunk 4]
```diff
@@ -52,8 +52,9 @@ public static class DateTimeDefinitions
 		public static readonly string WeekOfYearRegex = $@"(?<woy>(the\s+)?(?<cardinal>first|1st|second|2nd|third|3rd|fourth|4th|fifth|5th|last)\s+week(\s+of)?\s+({YearRegex}|{RelativeRegex}\s+year))";
 		public static readonly string FollowedDateUnit = $@"^\s*{DateUnitRegex}";
 		public static readonly string NumberCombinedWithDateUnit = $@"\b(?<num>\d+(\.\d*)?){DateUnitRegex}";
-		public static readonly string QuarterRegex = $@"(the\s+)?(?<cardinal>first|1st|second|2nd|third|3rd|fourth|4th)\s+quarter(\s+of|\s*,\s*)?\s+({YearRegex}|{RelativeRegex}\s+year)";
-		public static readonly string QuarterRegexYearFront = $@"({YearRegex}|{RelativeRegex}\s+year)\s+(the\s+)?(?<cardinal>first|1st|second|2nd|third|3rd|fourth|4th)\s+quarter";
+		public const string QuarterTermRegex = @"(((?<cardinal>first|1st|second|2nd|third|3rd|fourth|4th)\s+quarter)|(Q(?<number>[1-4])))";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: Can you also add support for H1/H2 for first and second halves of an year?

File: Specs/DateTime/English/DatePeriodParser.json (hunk 3 above)
Comment (tellarin) [reply Specs/DateTime/English/DatePeriodParser.json]: Also make the samples use all quarters. Not only Q3. :)

File: .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs (hunk 4 above)
Comment (STayinloves) [reply .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: OK

**Pull request merged**